src/views/Register/index.tsx

```diff
@@ -0,0 +1,306 @@
+    captcha?:string;
+}
+
+function getPasswordMatchCondition(referenceVal: string | undefined) {
+    return (val: string | undefined) => (
+        isTruthyString(val) && isTruthyString(referenceVal) && val !== referenceVal
+            ? 'Passwords do not match'
+            : undefined
+    );
+}
+
+const organizationTypes = [
+    { id: '101', key: 'NTLS', value: 'National Society' },
+    { id: '102', key: 'NGO', value: 'Non-Governmental Organization' },
+    { id: '103', key: 'UN', value: 'United Nations Agency' },
+];
+const nationalSocietyOptions = [
+    { id: '201', society_name: 'Red Cross Society' },
+    { id: '202', society_name: 'Red Crescent Society' },
+];
+const countryOptions = [
+    { id: '301', value: 'United States' },
+    { id: '302', value: 'Canada' },
+    { id: '303', value: 'United Kingdom' },
+];
+const whitelistedDomains = [
+    'example.com',
+    'anotherdomain.org',
+    'somedomain.net',
+];
+
+type FormFields = DefaultFormValue;
+
+const keySelector = (option: { id?: string }): string => option.id || '';
+const labelSelector = (option: { value?: string }): string => option.value || '';
+
+type PartialFormFields = Partial<FormFields>;
+type FormSchema = ObjectSchema<PartialFormFields>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const isWhitelistedEmail = (email: string): boolean => {
+    const domain = email.split('@')[1];
+    return whitelistedDomains.includes(domain);
+};
+
+const emailWhitelistValidation = (value: string | undefined) => {
+    if (!isWhitelistedEmail(value || '')) {
+        return 'Email not allowed';
+    }
+    return undefined;
+};
+
+const formSchema: FormSchema = {
+    fields: (value): FormSchemaFields => {
+        let fields: FormSchemaFields = {
+            first_name: {
+                required: true,
+                requiredValidation: requiredStringCondition,
+            },
+            last_name: {
+                required: true,
+                requiredValidation: requiredStringCondition,
+            },
+            email: {
+                required: true,
+                requiredValidation: requiredStringCondition,
+                validations: [emailCondition, emailWhitelistValidation],
+            },
+            password: {
+                required: true,
+                requiredValidation: requiredStringCondition,
+            },
+            confirm_password: {
+                required: true,
+                requiredValidation: requiredStringCondition,
+                forceValue: undefinedValue,
+                validations: [getPasswordMatchCondition(value?.password)],
+            },
+            organization: {
+                required: true,
+                requiredValidation: requiredStringCondition,
+            },
+            organization_type: {
+                required: true,
+            },
+            country: {
+                required: true,
+            },
+            city: {
+                required: true,
```

Comment:
@roshni73 , can you confirm if these are required?
Based on what I discussed with @barshathakuri , these are not required.